Update build and dependencies

abought opened the pull request:
# Purpose
Update build and dependency rules

- Fix missing files required to build
- Harmonize linting rules with other LZ projects- especially strict `eqeqeq`
- Remove gulp in favor of using webpack directly (tends to work better with IDE debugging tools)

abought commented:
Conversion to pure webpack is all finalized. Broken out into commit to reduce the line noise of finding changes.
https://github.com/statgen/raremetal.js/pull/12

The practical reason for this is to integrate more easily with IDE devtools like breakpoint debugger and code coverage, which should help with writing good unit tests. Getting those features to work with gulp was sometimes a bit fiddly in the past.

Passing to @welchr  for merge review: I didn't rerun the cli tools to verify function and I'd like to make sure that those behave for Alan before he starts work.

welchr commented:
Thanks Andy. Will be nice to be rid of gulp. 👍 

It looks like there's a test for the CLI but it is lacking. I will try to add a little more there, and also some docs about how to use the CLI too.

welchr commented:
I updated the CLI test case to cover the other tests, and added a little more to the docs about how to use the CLI as well. 

Can you rebase your branch on top of master and see if the new tests pass?

abought commented:
The extra CLI notes look like a really nice addition for developers. 

Branch rebased, tests are passing locally. Any other requests before merging?

welchr commented on src/app/fio.js:
```diff
@@ -186,7 +185,7 @@ class ScoreStatTable {
     let freq = this.altFreq[this.variantMap.get(variant)];
     if (freq == null) {
-      throw new Error("Variant did not exist when looking up alt allele freq: " + variant);
+      throw new Error(`Variant did not exist when looking up alt allele freq: ${  variant}`);
```
It seems like the new linting sometimes changes template strings to `${ variable}` and other times to `${ variable }`. Any idea what might be causing that?

abought replied:
Honestly not sure- I suspect it's a deeply arcane quirk of whatever AST transformations are used. This happened a bunch of times when I converted LocusZoom as well. Kind of annoying to clean up- once review is done I'll do a pass to standardize all of that.

Readability aside, using template strings should help avoid some of the [small quirks](https://stackoverflow.com/a/16522695/1422268) of the language around operator overloading/ "helpful" type coercion when writing pure JS.

abought commented:
Oh, right. *This* this the string-concatenation example I was looking for.
https://www.destroyallsoftware.com/talks/wat

(Someday I hope that the language will evolve to the point that I don't feel obliged to watch this once a year)

More seriously: 
- I've updated the eslint rules to fix this issue, and auto-applied the fix (`npx eslint src/ test/ --fix`)
- I've made several formatting rules less prescriptive, and added several "best practices" rules for common errors/ problems
  - "warn" level rules won't cause CI to fail, but they will allow the `--fix` mode to apply auto-formatting on demand.

All intended changes from me are complete in this branch; Ryan, any last requests before I merge?

welchr commented:
Awesome, thanks Andy. Looks much better. 

Go ahead and merge, just make sure to rebase and merge instead of creating a merge commit. I want to try and keep a clean linear history if possible. I'll turn on the branch protection rule for it as well to make sure Alan doesn't accidentally run into that.
